[Bug]: lint-fix workflow is failing

### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Describe the bug and reproduction steps

the lint-fix workflow is failing with the following error. We should remove the `--all-files` flag and just use `--files`.

```
Run pre-commit run --files openhands/**/* evaluation/**/* tests/**/* --config ./dev_config/python/.pre-commit-config.yaml --all-files
  pre-commit run --files openhands/**/* evaluation/**/* tests/**/* --config ./dev_config/python/.pre-commit-config.yaml --all-files
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.1[2](https://github.com/All-Hands-AI/OpenHands/actions/runs/11864077995/job/33066802842#step:8:2).7/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.12.7/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/[3](https://github.com/All-Hands-AI/OpenHands/actions/runs/11864077995/job/33066802842#step:8:3).12.7/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.7/x6[4](https://github.com/All-Hands-AI/OpenHands/actions/runs/11864077995/job/33066802842#step:8:4)
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.7/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.12.7/x[6](https://github.com/All-Hands-AI/OpenHands/actions/runs/11864077995/job/33066802842#step:8:6)4/lib
usage: pre-commit run [-h] [--color {auto,always,never}] [-c CONFIG]
                      [--verbose] [--all-files | --files [FILES ...]]
                      [--show-diff-on-failure]
                      [--hook-stage {commit-msg,post-checkout,post-commit,post-merge,post-rewrite,pre-commit,pre-merge-commit,pre-push,pre-rebase,prepare-commit-msg,manual}]
                      [--remote-branch REMOTE_BRANCH]
                      [--local-branch LOCAL_BRANCH] [--from-ref FROM_REF]
                      [--to-ref TO_REF]
                      [--pre-rebase-upstream PRE_REBASE_UPSTREAM]
                      [--pre-rebase-branch PRE_REBASE_BRANCH]
                      [--commit-msg-filename COMMIT_MSG_FILENAME]
                      [--prepare-commit-message-source PREPARE_COMMIT_MESSAGE_SOURCE]
                      [--commit-object-name COMMIT_OBJECT_NAME]
                      [--remote-name REMOTE_NAME] [--remote-url REMOTE_URL]
                      [--checkout-type CHECKOUT_TYPE]
                      [--is-squash-merge IS_SQUASH_MERGE]
                      [--rewrite-command REWRITE_COMMAND]
                      [hook]
pre-commit run: error: argument --all-files/-a: not allowed with argument --files
Error: Process completed with exit code 2.
```

Example: https://github.com/All-Hands-AI/OpenHands/actions/runs/11864077995/job/33066802842?pr=5058

### OpenHands Installation

Docker command in README

### OpenHands Version

_No response_

### Operating System

None

### Logs, Errors, Screenshots, and Additional Context

_No response_

--- a/.github/workflows/lint-fix.yml
+++ b/.github/workflows/lint-fix.yml
@@ -43,7 +43,7 @@ jobs:
         run: pip install pre-commit==3.7.0
       - name: Fix python lint issues
         run: |
-          pre-commit run --files openhands/**/* evaluation/**/* tests/**/* --config ./dev_config/python/.pre-commit-config.yaml --all-files
+          pre-commit run --files openhands/**/* evaluation/**/* tests/**/* --config ./dev_config/python/.pre-commit-config.yaml
 
       # Commit and push changes if any
       - name: Check for changes

